### PR TITLE
DNS configuration guidelines

### LINKED ISSUE #235: Document CNAME
### 📝 Description of the feature

Document valid CNAMES for pyansys.com

Right now we have the informal standard of docs.<product/service/library>.pyansys.com and we have great documentation of this at [DNS configuration](https://dev.docs.ansys.com/contents/ansys_best_practices/code_management/github/github_pages.html#dns-configuration)

Let's add part of that here within our dev documentation so we're consistent, and also clearly spell out who's responsible for adding the CNAME.

### DNS Takeover
Document the steps to take a DNS takeover.

CC @phxnsharp 

### 💡 Steps for implementing the feature

Reuse part of [DNS configuration](https://dev.docs.ansys.com/contents/ansys_best_practices/code_management/github/github_pages.html#dns-configuration).

### 🔗 Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by RobPasMue**
Closes #235

### CODE REVIEW

**Comment by RobPasMue** [comment]
Hi @PipKat - Could you please review this PR whenever possible? These are some high priority guidelines that we have been requested by Ryan Thomas in regards to CNAME convention and DNS takeover protection.

**Comment by RobPasMue** [comment]
@phxnsharp @akaszynski since you directly talked with Ryan Thomas for the resolution of this issue, could you please go through the docs and ensure that all the points have been treated accordingly? 

For documentation on how to verify domains on GH pages, I just redirected to GH docs. They'll be kept more up-to-date than our docs for sure.

File: doc/source/how-to/dns-configuration.rst
Comment (MaxJPRey) [inline doc/source/how-to/dns-configuration.rst:14]: Maybe here we could specify that in one hand the CNAME is located the `gh-pages` branch.
But on the other hand the Custom domain is also mentioned in GitHub repository `Settings/Pages `.

Talking about those two distinct locations where the custom domain is mentioned might help to clarify the process because I noticed some confusion for internal users.

Maybe not the perfect location to talk about it but it might be helpful.

Comment (RobPasMue) [reply doc/source/how-to/dns-configuration.rst:14]: CNAME configuration is done through Settings/Pages --> Custom domain.

Once you do that, GitHub automagically creates the ``CNAME`` file in your ``gh-pages`` branch without you having to do anything... Or in whatever branch your docs are published. Remember that the naming convention ``gh-pages`` is only that, a convention. You could potentially publish your docs under a branch called ``my-docs-branch``, for example.

I don't know if it's worth mentioning that the CNAME is also located in the ``gh-pages`` branch or not honestly.. because admins won't have to do anything there. It's done for them when you configure it in the Settings/Pages site. And if you change it, it also gets updated in the ``gh-pages`` branch.

But I'm open to discussion honestly 😄

File: doc/source/how-to/dns-configuration.rst
Comment (jorgepiloto) [inline doc/source/how-to/dns-configuration.rst:9]:
```suggestion
To request a CNAME for the ``pyansys.com`` domain, please contact
```

Comment (jorgepiloto) [reply doc/source/how-to/dns-configuration.rst:9]: I tend to use "in order to..." too. However, as per @PipKat reviews, I know most of the times its usage can be avoided as it does not introduce any information.

File: doc/source/how-to/dns-configuration.rst
Comment (jorgepiloto) [inline doc/source/how-to/dns-configuration.rst:14]:
```suggestion
documentation in GitHub pages to be exposed through it. To configure the CNAME
```

File: doc/source/how-to/dns-configuration.rst
Comment (jorgepiloto) [inline doc/source/how-to/dns-configuration.rst:21]: Suggesting to use `PyAnsys GitHub organization` here, as this term appears in other parts of the guide:

```suggestion
In the `PyAnsys GitHub organization`_, the following domains have been verified:
```

File: doc/source/how-to/dns-configuration.rst
Comment (jorgepiloto) [inline doc/source/how-to/dns-configuration.rst:53]: WHat about rewriting this as to avoid using "take" twice? 

```suggestion
Any attempt by external users/organizations of taking over the previous CNAME is identifiead and rejected by GitHub.
```

**Review by jorgepiloto — CHANGES_REQUESTED** [review]

**Review by jorgepiloto — APPROVED** [review]
Just left two minor suggestions but this looks good to me.

File: doc/source/how-to/dns-configuration.rst
Comment (MaxJPRey) [reply doc/source/how-to/dns-configuration.rst]: You are correct, thinking twice about it, it could bring more confusion.

**Review by MaxJPRey — APPROVED** [review]

File: doc/source/how-to/dns-configuration.rst
Comment (germa89) [inline doc/source/how-to/dns-configuration.rst:10]:
```suggestion
`Maxime Rey_`_, `Roberto Pastor Muela_`_ or `Alex Kaszynski_`_. Any of these members
```

File: doc/source/how-to/dns-configuration.rst
Comment (phxnsharp) [reply doc/source/how-to/dns-configuration.rst]: Also, you can publish now with a CI action without having a branch at all.

**Review by phxnsharp — APPROVED** [review]
If I were going to look for these naming guidelines, I would expect to find them in the "style guide" sections, not in the howto section. I think the changes are good, but there should also be something in the style guides. 

Also, we need an equivalent set of documentation over in the dev.docs.ansys.com site, since the problem was first found there and someone working over there might never come look at this site.

Thank you!

**Comment by phxnsharp** [comment]
> Also, we need an equivalent set of documentation over in the dev.docs.ansys.com site, since the problem was first found there and someone working over there might never come look at this site.

How long a day it has been. I already [created the issue to mirror the content](https://github.com/ansys/ansys-dev/issues/94) over there.

File: doc/source/how-to/dns-configuration.rst
Comment (RobPasMue) [reply doc/source/how-to/dns-configuration.rst]: Yep, that's also true. GH evolves fast 😄

**Comment by RobPasMue** [comment]
> If I were going to look for these naming guidelines, I would expect to find them in the "style guide" sections, not in the howto section. I think the changes are good, but there should also be something in the style guides.
> 

That's a good point, I can add something to the ``Documentation style`` section. :) 

> Also, we need an equivalent set of documentation over in the dev.docs.ansys.com site, since the problem was first found there and someone working over there might never come look at this site.
> 

Yes, I assigned myself that very same issue you mention. Once this gets approved and merged, I'll mirror the information to the other site. Thanks @phxnsharp!

**Comment by RobPasMue** [comment]
I'd like to have all reviewer approvals before merging since this is an important issue. Pinging @akaszynski and @PipKat for whenever they can review it.

File: doc/source/doc-style/index.rst
Comment (PipKat) [inline doc/source/doc-style/index.rst:51]:
```suggestion
For guidelines on how to acquire a specific DNS for hosting your documentation,
see :ref:`DNS configuration`. You should ensure that you are compliant with
the naming convention for your CNAME.
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:10]: Given that this guide is public-facing, do we want to use specific names or use the PyAnsys support email address?

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:9]:
```suggestion
To request a CNAME for the ``pyansys.com`` domain, contact
```
Avoid the use of "please" in technical documentation--according to all style guides!

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:16]:
```suggestion
for your documentation, refer to `Managing a custom domain for your GitHub Pages site`_.
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:21]:
```suggestion
In the `PyAnsys GitHub organization`_, these domains have been verified:
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:36]:
```suggestion
As explained in `Verifying a domain for your organization site`_, GitHub provides for
verifying domains for users and organizations.
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:46]: I'm changing the lists to bullets as these aren't procedural (ordered) steps.

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:48]:
```suggestion
- Consider that the domain ``pyansys.com`` has been verified for the `PyAnsys GitHub organization`_.
- This CNAME is requested: ``subdomain.pyansys.com``.
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:51]:
```suggestion
This CNAME can only be used by repositories inside the `PyAnsys GitHub organization`_.
Any attempt by an external user to take over this CNAME is identified and rejected by GitHub.
```
Avoid slashes when possible. Pick the most descriptive choice.

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:60]:
```suggestion
- The domain ``pyansys.com`` has been verified for the `PyAnsys GitHub organization`_.
- This CNAME is requested: ``subsubdomain.subdomain.pyansys.com``.

This CNAME **can** be used by external users for their repositories. For this reason,
you must avoid creating CNAME requests that are not verified by the organization.
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:67]:
```suggestion
CNAME values have been taken over in the past by external users, typically due to these reasons:
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:72]:
```suggestion
* Long time lapses occurred between CNAME creation and assignment to GitHub pages.
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:74]:
```suggestion
Thus, it is important that you follow these guidelines:
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:77]:
```suggestion
* Check that the CNAME that you request does not have a subdomain depth larger than **1** with respect to the verified domains.
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:78]:
```suggestion
* Request a CNAME only when needed, which is just prior to publishing the site.
```

File: doc/source/how-to/dns-configuration.rst
Comment (PipKat) [inline doc/source/how-to/dns-configuration.rst:79]:
```suggestion
* Request deletion of the CNAME once it is no longer used to prevent others from hosting their sites on it.
```

File: doc/source/how-to/dns-configuration.rst
Comment (RobPasMue) [reply doc/source/how-to/dns-configuration.rst:9]: Thanks Kathy!!

File: doc/source/how-to/dns-configuration.rst
Comment (RobPasMue) [reply doc/source/how-to/dns-configuration.rst:10]: This specific part of the documentation, even if it's public facing, is directed for Ansys employees only, since they are the ones that can request CNAME creations. I think it's fine if we leave it as is.

File: doc/source/how-to/dns-configuration.rst
Comment (RobPasMue) [reply doc/source/how-to/dns-configuration.rst:46]: Sure, thanks @PipKat!!

**Comment by RobPasMue** [comment]
Thanks all for your thorough reviews. Merging!

**Comment by RobPasMue** [comment]
GraphViz complaining and Azure servers for ubuntu down... We've got to wait before merging.

**Pull request merged**
